ApiDoctor.Console/Program.cs

```diff
@@ -1960,14 +1963,139 @@ private static async Task<bool> GenerateSnippetsAsync(GenerateSnippetsOptions op
+            if (!File.Exists(snippetNotAvailableFileFullName))
+            {
+                const string includeSnippetNotAvailableText = "```\r\nSnippet not available\r\n```";
+                File.WriteAllText(snippetNotAvailableFileFullName, includeSnippetNotAvailableText);
+            }
+            return snippetsTabSectionForMethod.ToString();
+        }
+
+        private static string GetSnippetContentForMethodByLanguage(string language, string snippetPrefix, string snippetsPath)
+        {
+            var fileName = $"{snippetPrefix}---{language.ToLowerInvariant()}";
+            var fileFullPath = Path.Combine(snippetsPath, fileName);
+
+            FancyConsole.WriteLine(FancyConsole.ConsoleSuccessColor, $"Reading {fileFullPath}");
+
+            string codeSnippet = null;
+            if (File.Exists(fileFullPath))
+                codeSnippet = File.ReadAllText(fileFullPath);
+            else
+                FancyConsole.WriteLine(FancyConsole.ConsoleErrorColor, $"Error: file '{fileName}' does not exist");
+
+            return codeSnippet;
+        }
+
+        private static Dictionary<string, HashSet<string>> GetSnippetsLanguageSetForDocSet(DocFile[] docFiles, string[] languageOptions, string snippetsPath)
+        {
+            var snippetsLanguageSetBySourceFile = new Dictionary<string, HashSet<string>>();
+            var snippetTempFiles = Directory.EnumerateFiles(snippetsPath);
+            foreach (DocFile docFile in docFiles)
+            {
+                var languages = new HashSet<string>();
+                foreach (var request in docFile.Requests)
+                {
+                    // no need to continue if we already have the full list of languages
+                    if (languages.Count == languageOptions.Length)
+                        break;
+
+                    string snippetPrefix;
+                    try { snippetPrefix = GetSnippetPrefix(request); } catch (ArgumentException) { continue; }
+
+                    var expectedFileName = $"{Path.Combine(snippetsPath, snippetPrefix)}---";
+                    var snippetLanguagesForMethod = snippetTempFiles
+                        .Where(x => x.StartsWith(expectedFileName))
+                        .Select(x => x.Substring(expectedFileName.Length))
+                        .ToHashSet();
+                    if (!snippetLanguagesForMethod.Any())
+                    {
+                        snippetsLanguageSetBySourceFile.Add($"{request.SourceFile.DisplayName}/{request.Identifier}", snippetLanguagesForMethod);
+                        continue;
+                    }
+
+                    languages.UnionWith(snippetLanguagesForMethod);
+                }
+
+                // just so that we have the correct casing displayed as tab names in docs e.g. PowerShell and not powershell
+                languages = languageOptions.Where(x => languages.Contains(x, StringComparer.OrdinalIgnoreCase)).ToHashSet();
```

Comment:
```suggestion
                languages = languageOptions.Where(x => languages.Contains(x)).ToHashSet();
```

Reply:
Move the comparison parameter to the constructor. (see other comment)
https://stackoverflow.com/questions/2667635/how-to-use-hashsetstring-contains-method-in-case-insensitive-mode